pkg/controller/direct/bigquery/table_controller.go

```diff
@@ -239,32 +255,61 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 	a.customTableLogic(table)
 	parent := a.id.Parent()
 
-	// TODO: Update annotation logic after confirming requirement.
-	// if a.desired.ObjectMeta.Annotations != nil {
-	// 	unmanaged, ok := a.desired.ObjectMeta.Annotations["cnrm.cloud.google.com/unmanaged"]
-	// 	if ok && unmanaged != "" {
-	// 		unmanagedFields := strings.Split(unmanaged, ",")
-	// 		for _, field := range unmanagedFields {
-	// 			// This ability is only intended for spec.schema field for the moment.
-	// 			if field == "spec.schema" {
-	// 				table.Schema = nil
-	// 			}
-	// 		}
-	// 		// Make PATCH call with nil schema to avoid schema being updated.
-	// 		res, err := a.gcpService.Patch(parent.ProjectID, parent.DatasetID, a.id.ID(), table).Do()
-	// 		if err != nil {
-	// 			return fmt.Errorf("error updating Table %s: %w", a.id.ID(), err)
-	// 		}
-	// 		return a.UpdateStatusForUpdate(ctx, updateOp, res)
-	// 	}
-	// }
+	if len(a.unmanagedFields) > 0 {
+		// Make PATCH call without unmanaged fields to avoid accidental override
+		res, err := a.gcpService.Patch(parent.ProjectID, parent.DatasetID, a.id.ID(), table).Do()
+		if err != nil {
+			return fmt.Errorf("error updating Table %s: %w", a.id.ID(), err)
+		}
+		return a.UpdateStatusForUpdate(ctx, updateOp, res)
+	}
 	res, err := a.gcpService.Update(parent.ProjectID, parent.DatasetID, a.id.ID(), table).Do()
```

Comment:
Nit: Could you add a comment here that we make Update() call here for backwards compatibility purposes?